src/IR/IR.ts

```diff
@@ -271,7 +272,7 @@ export interface BinaryOp {
  */
 export interface MutatingBinaryOp {
   type: "MutatingBinaryOp";
-  op: BuiltinBinop;
+  op: string;
```

Comment:
Why allow an arbitrary string here? I find it's better to know that you're limited to a certain set, even if every language doesn't use that whole set

Reply:
I did it to make BinaryOp -> MutatingBinaryOp work, since BinaryOp.op was arbitrary string. I agree but those two nodes need to match. The plugins should not rename the ops, emitters can handle that.